demo-app/src/lib/mysql-uqi-client.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Iterator,
+  uqi,
+  type UqiClient,
+  type UqiColumn,
+  type UqiContext,
+  type UqiMappedType,
+  type UqiResult,
+} from '@open-truss/open-truss'
+import mysql, { type Connection, type FieldPacket } from 'mysql2/promise'
+import { URL } from 'url'
+
+export interface MysqlConfig {
+  uri: string
+  socketPath?: string
+}
+
+function makeUqiColumnCompatible(fields: FieldPacket[]): UqiColumn[] {
```

Comment:
Maybe a comment here?

```suggestion
// mysql2's FieldPacket.type is a number but UqiColumn.type is a string so we map
// the number to a string.
function makeUqiColumnCompatible(fields: FieldPacket[]): UqiColumn[] {
```